File: Searching/BinarySearch.py
# O(log(n)) time | O(log(n)) space
def binarySearch(array, target):
    # Write your code here.
    # pass
	# define variable: start, middle, & end
	lo = 0
	hi = len(array) - 1
	# mid = (hi - lo) // 2
	
	# start = array[0]
	# end = array[len(array)]
	# middle = (end - start) / 2
	# while start does not equal end (for loop)
	# for idx, item in enumerate(array):
	while (lo <= hi):
		mid = lo + (hi - lo) // 2
		if target < array[mid]:
			hi = mid -1
		elif target > array[mid]:
			lo = mid + 1
		else:
			return mid
	
		
	# if start equals target, return index
	return -1

File: Searching/BinarySearch.java
import java.util.*;

class Program {
	// O(log(n)) time || O(1) space
  public static int binarySearch(int[] array, int target) {
    // Write your code here.
		// create lo, high, and mid variables
		// While lo <= high run the following algo:
			// SET MID??
			// if target > mid, then set lo equal to mid + 1. 
			// if target < mid, then set hi equalt to mid - 1
			// if target equals value at mid index, return mid index
		int lo = 0;
		int hi = array.length - 1; 
		
		
		while ( lo <= hi ) {
			int mid = lo + (( hi - lo ) / 2);
			if (target > array[mid] ) {
				lo = mid + 1;
			} else if ( target < array[mid] ) {
				hi = mid - 1;
			} else if (target == array[mid] ) {
				return mid;
			}
		}
    return -1;
  }
}
